### PR TITLE
[kube-prometheus-stack] Add support for deploying blackbox-exporter.

### PR DESCRIPTION
**Pull request by tiagomeireles**
#### What this PR does / why we need it:
Add support for deploying blackbox exporter. Upstream [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus) has this feature. 

This PR still needs a bit more work but i'd like to confirm that this would be accepted before I invest the time.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by bismarck** [comment]
@tiagomeireles You need to [sign](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) your commits.

**Comment by tiagomeireles** [comment]
@bismarck signed + rebased of HEAD.

**Comment by tiagomeireles** [comment]
Bump..

**Comment by bismarck** [comment]
@tiagomeireles Not sure why this has to be bundled with the kube-prometheus-stack? You can deploy the blackbox-exporter separately.

@vsliouniaev thoughts?

File: charts/kube-prometheus-stack/templates/exporters/blackbox-exporter/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{{- if .Values.blackboxExporter.enabled }}
```
Comment (bismarck) [inline charts/kube-prometheus-stack/templates/exporters/blackbox-exporter/servicemonitor.yaml:1]: The blackbox-exporter chart already [defines](https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-blackbox-exporter/templates/servicemonitor.yaml) a service monitor. Why is this needed?

**Review by bismarck — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/values.yaml [hunk 2]
```diff
@@ -1184,6 +1184,52 @@ kube-state-metrics:
   podSecurityPolicy:
     enabled: true
 
+## Blackbox Exporter configuration
+##
+blackboxExporter:
+  enabled: true
```
[the remark this answers is not in the document]
Comment (tiagomeireles) [reply charts/kube-prometheus-stack/values.yaml:1209]: The example [jsonnet](https://github.com/prometheus-operator/kube-prometheus/blob/main/example.jsonnet) and [kustomization.yaml](https://github.com/prometheus-operator/kube-prometheus/blob/main/kustomization.yaml) have it enabled. I followed that. Im happy to flip the default.

File: charts/kube-prometheus-stack/templates/exporters/blackbox-exporter/servicemonitor.yaml (hunk 1 above)
Comment (tiagomeireles) [reply charts/kube-prometheus-stack/templates/exporters/blackbox-exporter/servicemonitor.yaml:1]: As does other exporters packaged with kube-prometheus-stack. e.g. [node-exporter](https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-node-exporter/templates/monitor.yaml) 

I followed the existing pattern.

**Comment by tiagomeireles** [comment]
> @tiagomeireles Not sure why this has to be bundled with the kube-prometheus-stack? You can deploy the blackbox-exporter separately.

We could say the same about all the components deployed in kube-prometheus-stack. 

This component is available in kube-prometheus which AFAIK this chart loosely follows. This change would make [probes](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/crds/crd-probes.yaml) CR available out of the box.

File: charts/kube-prometheus-stack/values.yaml [hunk 3]
```diff
@@ -1203,6 +1203,52 @@ kube-state-metrics:
   podSecurityPolicy:
     enabled: true
 
+## Blackbox Exporter configuration
+##
+blackboxExporter:
```
Comment (maximbaz-maersk) [inline charts/kube-prometheus-stack/values.yaml:1208]: Would it be possible to support `namespaceOverride` like other subcharts?

Thanks for this PR by the way, looking forward to see it merged!

Comment (tiagomeireles) [reply charts/kube-prometheus-stack/values.yaml:1208]: This will require added it to the subchart. Happy to do so assuming this is a blocker.

File: charts/kube-prometheus-stack/templates/exporters/blackbox-exporter/servicemonitor.yaml (hunk 1 above)
Comment (bismarck) [reply charts/kube-prometheus-stack/templates/exporters/blackbox-exporter/servicemonitor.yaml:1]: The service monitor is already defined in the blackbox exporter chart. It doesn't need to also be added to kube-prometheus-stack.
